tests/integration/local/invoke/invoke_cdk_templates.py

```diff
@@ -15,13 +16,122 @@
 from tests.testing_utils import IS_WINDOWS, RUNNING_ON_CI, CI_OVERRIDE
 
 
+class TestCDKSynthesizedTemplate(InvokeIntegBase):
+    template = Path("cdk/cdk_template.yaml")
+
+    @pytest.mark.flaky(reruns=3)
+    def test_invoke_returncode_is_zero(self):
+        command_list = self.get_command_list(
+            "helloworld-serverless-function", event_path=self.event_path, template_path=self.template_path
+        )
+        stdout, _, return_code = self.run_command(command_list)
+
+        self.assertEqual(return_code, 0)
+
+    @pytest.mark.flaky(reruns=3)
+    def test_invoke_with_utf8_event(self):
+        command_list = self.get_command_list(
+            "helloworld-serverless-function", event_path=self.event_utf8_path, template_path=self.template_path
+        )
+        stdout, _, return_code = self.run_command(command_list)
+
+        self.assertEqual(return_code, 0)
+
+    @pytest.mark.flaky(reruns=3)
+    def test_invoke_returns_expected_results(self):
+        command_list = self.get_command_list(
+            "helloworld-serverless-function", event_path=self.event_path, template_path=self.template_path
+        )
+
+        stdout, _, return_code = self.run_command(command_list)
+
+        process_stdout = stdout.strip()
+        self.assertEqual(process_stdout.decode("utf-8"), '"Hello world"')
+
+    @pytest.mark.flaky(reruns=3)
+    def test_invoke_with_timeout_set(self):
+        command_list = self.get_command_list(
+            "timeout-function", event_path=self.event_path, template_path=self.template_path
+        )
+
+        start = timer()
+
+        stdout, _, return_code = self.run_command(command_list)
+
+        end = timer()
+
+        wall_clock_cli_duration = end - start
+
+        process_stdout = stdout.strip()
+
+        # validate the time of the cli (timeout is set to 5s)
+        self.assertGreater(wall_clock_cli_duration, 5)
+        self.assertLess(wall_clock_cli_duration, 20)
+
+        self.assertEqual(return_code, 0)
+        self.assertEqual(
```

Comment:
Looking at the function, it only returns after 10 seconds, which means the function is not timed out but finished running. Should we add a timeout case, and rename this one?

Reply:
Fixing the timeout value you pointed out should fix this issue.